### PR TITLE
Reduce responsibilities of TypeResolver

### PR DESCRIPTION
**Pull request by torbsto**
Before this commit, the TypeResolver's interface comprised:
- fromString
- fromObject
- toJSONString
- elementType

This is now reduced to only fromString.

fromObject was completely removed, toJsonString is done by Jackson. The elementType interface was required for deserialization with Jackson. Since this is equivalent to TypeResolver's fromString, it was replaced with it.

### CODE REVIEW

File: common/src/test/java/com/bakdata/quick/common/api/client/TopicDataClientTest.java [hunk 1]
```diff
@@ -44,7 +45,7 @@ class TopicDataClientTest {
     private final String host = String.format("%s:%d", this.server.getHostName(), this.server.getPort());
     private final MirrorHost mirrorHost = new MirrorHost(this.host, MirrorConfig.directAccess());
     private final MirrorClient<String, TopicData> topicDataClient =
-        new DefaultMirrorClient<>(this.mirrorHost, this.client, this.mapper.constructType(TopicData.class));
+        new DefaultMirrorClient<>(this.mirrorHost, this.client, new KnownTypeResolver<>(TopicData.class, this.mapper));
```
Comment (raminqaf) [inline common/src/test/java/com/bakdata/quick/common/api/client/TopicDataClientTest.java:48]: How about we change the name of this test? It was pretty misleading the first time I looked at it. For example `TopicRegistryMirrorClientTest` or similar to that.

File: common/src/main/java/com/bakdata/quick/common/type/QuickTopicType.java [hunk 2]
```diff
@@ -22,39 +22,110 @@
...
-
-    QuickTopicType(final Supplier<Serde<?>> serdeSupplier, final Supplier<TypeResolver<?>> resolverSupplier) {
-        this.serdeSupplier = serdeSupplier;
-        this.resolverSupplier = resolverSupplier;
-    }
+    SCHEMA {
+        @Override
+        public <K> TypeResolver<K> getTypeResolver(@Nullable final ParsedSchema parsedSchema) {
+            Objects.requireNonNull(parsedSchema, "Schema must not be null for Avro types");
+            if (!(parsedSchema instanceof AvroSchema)) {
```
Comment (raminqaf) [inline common/src/main/java/com/bakdata/quick/common/type/QuickTopicType.java:43]: I guess later on we add another if-else for Protobuf support. Correct?

**Review by raminqaf — APPROVED** [review]
LGTM for the refactoring part! I had some small points and questions that I wrote in my reviews.

File: common/src/main/java/com/bakdata/quick/common/api/client/DefaultMirrorClient.java [hunk 3]
```diff
@@ -45,47 +43,45 @@
...
     private final HttpClient client;
-    private final JavaType valueType;
-    private final JavaType listType;
+    private final MirrorValueParser<V> parser;
 
     /**
-     * Constructor for client.
+     * Constructor for the client.
      *
-     * @param topicName name of the topic the mirror is deployed
```
Comment (DawidNiezgodka) [inline common/src/main/java/com/bakdata/quick/common/api/client/DefaultMirrorClient.java:54]: Maybe we'll add a preposition here, for example under which or for which the mirror is deployed

**Review by DawidNiezgodka — APPROVED** [review]

File: common/src/main/java/com/bakdata/quick/common/resolver/GenericAvroResolver.java [hunk 4]
```diff
@@ -38,38 +29,12 @@
 public class GenericAvroResolver implements TypeResolver<GenericRecord> {
 
     private final JsonAvroConverter converter;
-    private final JavaType elementType;
     @Nullable
     private Schema schema = null;
```
[the remark this answers is not in the document]
Comment (torbsto) [reply common/src/main/java/com/bakdata/quick/common/resolver/GenericAvroResolver.java:33]: Oh, true. I think this should even be final and not nullable anymore

File: common/src/main/java/com/bakdata/quick/common/type/QuickTopicType.java (hunk 2 above)
Comment (torbsto) [reply common/src/main/java/com/bakdata/quick/common/type/QuickTopicType.java:43]: Sort of. My idea is to deprecate `SCHEMA` and instead add `AVRO` and `PROTO`

File: common/src/main/java/com/bakdata/quick/common/resolver/KnownTypeResolver.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.UncheckedIOException;
+
+/**
+ * Type resolver for known types.
+ *
+ * @param <T> type to resolve
+ */
+public class KnownTypeResolver<T> implements TypeResolver<T> {
```
[the remark this answers is not in the document]
Comment (torbsto) [reply common/src/main/java/com/bakdata/quick/common/resolver/KnownTypeResolver.java:32]: Added some context

**Pull request merged**